dragonfly/implementations/psyllid_provider.py

```diff
@@ -182,12 +185,14 @@ def set_central_frequency(self, channel, cf):
             return True
 
         except core.exceptions.DriplineError:
-            logger.error('Could not set central frequency of Psyllid instance for channel {}'.format(channel))
-            self.freq_dict[channel]=None
-            return False
-        except Exception as e:
-            logger.error('Unexpected error in set_central_frequency')
-            raise 
+            if self.mode_testing == True:
+                self.freq_dict[channel]=None
+                logger.info('Could not set central frequency of Psyllid instance for channel {}'.format(channel))
```

Comment:
normally yes. 
but if self.mode_testing itsTrue it is not. Because then *get_acquisition_mode* in psyllid_interface is testing which settings can be applied to psyllid. Otherwise the failure to set the frequency results in a logger.critical message.

Reply:
is this testing mode something we want in the release? I understand this is useful for testing, but do we need it?

Reply:
we need it when psyllid_provider is testing whether psyllid is in streaming or triggering mode. it is generally set to false and then failing to set the frequency will result in an error.